react/components/ProductBrand/index.js

```diff
@@ -0,0 +1,177 @@
+import PropTypes from 'prop-types'
+import React, { useContext } from 'react'
+import { ProductContext } from 'vtex.product-context'
+import { generateBlockClass } from '@vtex/css-handles'
+import { Query } from 'react-apollo'
+import gql from 'graphql-tag'
+import { changeImageUrlSize } from './urlHelpers'
+
+import styles from './productBrand.css'
+
+const BRAND_LOGO_QUERY = gql`
+  query Brand($id: Int) {
+    brand(id: $id) {
+      imageUrl
+    }
+  }
+`
+
+const DISPLAY_MODE = {
+  LOGO: 'logo',
+  TEXT: 'text',
+}
+
+const shouldExcludeBrand = (brandName, brandId, excludeList) => {
+  if (Array.isArray(excludeList)) {
+    return excludeList.includes(brandName) || excludeList.includes(brandId)
+  }
+  return false
+}
+
+const ProductBrand = ({
+  displayMode = DISPLAY_MODE.LOGO,
+  fallbackToText = true,
+  loadingPlaceholder = DISPLAY_MODE.LOGO,
+  /** TODO: decide whether the prop width should be supported
+   * It makes sense at surface, but setting both width and height
+   * messes with the alignment of the logo, due to how our image
+   * server handles resizing. */
+  // width,
+  height = 100,
+  excludeBrands,
+  blockClass,
+}) => {
+  const productContext = useContext(ProductContext)
+
+  const {
+    product: { brand: brandName, brandId },
```

Comment:
How it's dealing when product is loading or error?

Reply:
by crashing and burning

Reply:
(updated)

Reply:
:rage1: